apps/testing-modal/src/app/app.component.spec.ts

```diff
@@ -1,16 +1,77 @@
-import { render } from '@testing-library/angular';
+import { fireEvent, render, screen } from '@testing-library/angular';
 import { AppComponent } from './app.component';
+import userEvent from '@testing-library/user-event';
 
 describe('AppComponent', () => {
-  test('error modal is displayed if you click on "Confirm" without inputing a name', async () => {
+  test('error modal is displayed if you click on "Confirm" without inputting a name 1', async () => {
     await render(AppComponent);
+
+    const button = screen.getByRole('button');
+
+    fireEvent.click(button);
+
+    expect(document.getElementsByTagName('h1')[0]).toBeInTheDocument();
+  });
+
+  test('error modal is displayed if you click on "Confirm" without inputting a name 2', async () => {
+    await render(AppComponent);
+
+    const button = screen.getByRole('button');
+
+    fireEvent.click(button);
+
+    const errorHeading = document.getElementsByTagName(
+      'h1'
+    )[0] as HTMLHeadingElement;
+
+    expect(errorHeading).toHaveTextContent('Error');
   });
 
   test('error message is shown if you click "Cancel" in the confirmation modal after submitting a name', async () => {
     await render(AppComponent);
+
+    const button = screen.getByRole('button');
+
+    const input = screen.getByRole('textbox');
+
+    await userEvent.type(input, 'William');
+
+    await userEvent.click(button);
+
+    const cancelButton = document.querySelectorAll(
+      'button'
+    )[1] as HTMLButtonElement;
```

Comment:
this is dangerous, if you change position or add a new button, your test will fail which is a false position. (or you really want your test to fail because number of buttons and position matter)

Otherwise, you can do this: `    const confirmDialogButton = screen.getByRole('button', {
          name: /confirm/i,
        });`